### DISCUSSION TITLE
Mergify seems to apply rule to all PRs and not just new PRs?

### DISCUSSION THREAD

**Discussion by cachedout** [Mergifyio/mergify#2412] [Q&A]
Hello!

I have a question that I can't seem to find in a cursory look through the documentation and I would like to better understand the behavior I just experienced from Mergify.

We enabled the Mergify integration on a repo and [defined a ruleset](https://github.com/elastic/package-storage/blob/master/.mergify.yml). We then submitted a test PR and Mergify did the right thing in monitoring the PR while its checks run and after determining that all checks passed, it merged the PR.

However, at the time that it merged the PR, it also merged all _other_ open PRs in the repo which matched the Mergify ruleset. Thankfully, this was not too much of an issue but it was surprising behavior. (At least, it was surprising to us!)

Can somebody help us understand whether this is the expected behavior and, if not, how this might have occurred? Thanks!

**Comment by jd** [comment]
That's indeed the expected behavior, the rules are applied on every (open) PR. Since you merged a PR Mergify went ahead and re-evaluated every PR that's why you saw it "so fast".
Anyhow, if any event (label, comment, ci) is triggered on an existing PR, it is evaluated by Mergify.

**Reply by cachedout** [reply]
Great! Thanks for the quick reply. This behavior was a bit surprising for us, but it's clear now why this would be the case. 

I might suggest that this be included in the documentation if it's not already. I looked but didn't see it. Anyhow, thanks again!
